lib/shopify_dashboard_plus.rb

```diff
@@ -102,35 +130,43 @@ def validate_date_range(from, to)
 
 
   ## Metrics Helpers
-  def get_total_revenue(orders)
-    revenue = orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) rescue 0
-    revenue ||= 0
+  def display_as_currency(value)
+    ShopifyAPI::Shop.current.money_with_currency_format.gsub("{{amount}}", value.to_s) rescue "N/A"
+  end
+
+  def get_date_range(first, last)
+    DateTime.parse(last).mjd - DateTime.parse(first).mjd
+  end
+
+  def get_total_revenue(collections)
+    # collections is an array of ActiveRecord::Collections, each containing orders
+    collections.collect{|orders| orders.collect{|order| order.total_price.to_f }.inject(:+).round(2) }.inject(:+).round(2) rescue 0
   end
   
-  def get_daily_revenues(start_date, end_date, orders)
+  def get_daily_revenues(start_date, end_date, collections)
     # Create hash entry for total interval over which to inspect sales
     revenue_per_day = {}
-    days = DateTime.parse(end_date).mjd - DateTime.parse(start_date).mjd
+    days = get_date_range(start_date, end_date)
     (0..days).each{ |day| revenue_per_day[(DateTime.parse(end_date) - day).strftime("%Y-%m-%d")] = 0 }
 
-    # Retreive orders between start and end date (up to 50)
-    revenue = orders.collect{|order| [order.created_at, order.total_price.to_f]}
+    # Retreive array of ActiveRecord::Collections, each containing orders between the start and end date
+    revenue = collections.collect{|orders| orders.collect{|order| [order.created_at, order.total_price.to_f]}}
     
     # Filter order details into daily totals and return
-    revenue.each do |sale|
-      revenue_per_day[DateTime.parse(sale[0]).strftime('%Y-%m-%d')] += sale[1]
+    revenue.each do |order|
+      order.each do |sale|
```

Comment:
Ruby has destructuring syntax so you can do

```ruby
order.each do |(date, total)|
  # stuff
end
```

instead of accessing things by index.

You can also do this when assigning variables

```ruby
a, b = [1,2]
a # 1
b # 2
```